### PR TITLE
netdata: fix build without cloud support

### PR DESCRIPTION
**Pull request by Izorkin**
## Description of changes
Fix build netdata without cloud support.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
> I think, it would be good to add as a top-level package `netdataNoCloud` which will disable cloud support (and cloud UI I assume) and be tested as part of nixpkgs.

To test the Cloud in `nixos\test`, Internet access is required. I don't know how to do this.

**Comment by RaitoBezarius** [comment]
> Allow this variant?
> 
> ```diff
> diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
> index f380bf4fd6f6..c76bcc7aa094 100644
> --- a/pkgs/tools/system/netdata/default.nix
> +++ b/pkgs/tools/system/netdata/default.nix
> @@ -2,13 +2,13 @@
>  , CoreFoundation, IOKit, libossp_uuid
>  , nixosTests
>  , netdata-go-plugins
> -, bash, curl, jemalloc, libuv, zlib, libyaml
> +, bash, curl, jemalloc, json_c, libuv, zlib, libyaml
>  , libcap, libuuid, lm_sensors, protobuf
>  , withCups ? false, cups
>  , withDBengine ? true, lz4
>  , withIpmi ? (!stdenv.isDarwin), freeipmi
>  , withNetfilter ? (!stdenv.isDarwin), libmnl, libnetfilter_acct
> -, withCloud ? (!stdenv.isDarwin), json_c
> +, withCloud ? false
>  , withCloudUi ? false
>  , withConnPubSub ? false, google-cloud-cpp, grpc
>  , withConnPrometheus ? false, snappy
> @@ -42,14 +42,13 @@ stdenv.mkDerivation rec {
> 
>    nativeBuildInputs = [ autoreconfHook pkg-config makeWrapper protobuf ];
>    # bash is only used to rewrite shebangs
> -  buildInputs = [ bash curl jemalloc libuv zlib libyaml ]
> +  buildInputs = [ bash curl jemalloc json_c libuv zlib libyaml ]
>      ++ lib.optionals stdenv.isDarwin [ CoreFoundation IOKit libossp_uuid ]
>      ++ lib.optionals (!stdenv.isDarwin) [ libcap libuuid ]
>      ++ lib.optionals withCups [ cups ]
>      ++ lib.optionals withDBengine [ lz4 ]
>      ++ lib.optionals withIpmi [ freeipmi ]
>      ++ lib.optionals withNetfilter [ libmnl libnetfilter_acct ]
> -    ++ lib.optionals withCloud [ json_c ]
>      ++ lib.optionals withConnPubSub [ google-cloud-cpp grpc ]
>      ++ lib.optionals withConnPrometheus [ snappy ]
>      ++ lib.optionals (withCloud || withConnPrometheus) [ protobuf ]
> diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
> index 9718a46468eb..6194c4fae07b 100644
> --- a/pkgs/top-level/all-packages.nix
> +++ b/pkgs/top-level/all-packages.nix
> @@ -10213,6 +10213,13 @@ with pkgs;
>      inherit (darwin.apple_sdk.frameworks) CoreFoundation IOKit;
>      protobuf = protobuf3_21;
>    };
> +
> +  netdataCloud = callPackage ../tools/system/netdata {
> +    inherit (darwin.apple_sdk.frameworks) CoreFoundation IOKit;
> +    protobuf = protobuf3_21;
> +    withCloud = true;
> +    withCloudUi = true;
> +  };
>    # Exposed here so the bots can auto-upgrade it
>    netdata-go-plugins = callPackage ../tools/system/netdata/go.d.plugin.nix { };
> ```

Sounds good actually, but please add a release notes to mention that we decided to remove the cloud and the cloud UI from the default `netdata`.

**Comment by Izorkin** [comment]
Update PR.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10213,6 +10213,10 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation IOKit;
     protobuf = protobuf3_21;
   };
+  netdataCloud = netdata.override {
+    withCloud = true;
```
Comment (RaitoBezarius) [inline pkgs/top-level/all-packages.nix:10217]: `!stdenv.isDarwin` should go here?

Comment (Izorkin) [reply pkgs/top-level/all-packages.nix:10217]: Fixed.

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 2]
```diff
@@ -336,6 +336,8 @@ The module update takes care of the new config syntax and the data itself (user
 
 - Suricata was upgraded from 6.0 to 7.0 and no longer considers HTTP/2 support as experimental, see [upstream release notes](https://forum.suricata.io/t/suricata-7-0-0-released/3715) for more details.
 
+- Cloud support is disabled by default in `netdata` package. To enable Cloud support need use `netdataCloud` package.
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2311.section.md:339]:
```suggestion
- Cloud support in the `netdata` package is now disabled by default. To enable it use the `netdataCloud` package.
```

Comment (Izorkin) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: Thanks, fixed.

File: pkgs/tools/system/netdata/default.nix [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
 , netdata-go-plugins
-, bash, curl, jemalloc, libuv, zlib, libyaml
+, bash, curl, jemalloc, json_c, libuv, zlib, libyaml
 , libcap, libuuid, lm_sensors, protobuf
 , withCups ? false, cups
 , withDBengine ? true, lz4
 , withIpmi ? (!stdenv.isDarwin), freeipmi
 , withNetfilter ? (!stdenv.isDarwin), libmnl, libnetfilter_acct
-, withCloud ? (!stdenv.isDarwin), json_c
+, withCloud ? false && (!stdenv.isDarwin)
```
Comment (RaitoBezarius) [inline pkgs/tools/system/netdata/default.nix:11]: this is always… false?

Comment (Izorkin) [reply pkgs/tools/system/netdata/default.nix]: Ups... Fixed.

File: pkgs/tools/system/netdata/default.nix [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 , netdata-go-plugins
-, bash, curl, jemalloc, libuv, zlib, libyaml
+, bash, curl, jemalloc, json_c, libuv, zlib, libyaml
 , libcap, libuuid, lm_sensors, protobuf
 , withCups ? false, cups
 , withDBengine ? true, lz4
 , withIpmi ? (!stdenv.isDarwin), freeipmi
 , withNetfilter ? (!stdenv.isDarwin), libmnl, libnetfilter_acct
-, withCloud ? (!stdenv.isDarwin), json_c
+, withCloud ? false || (false && (!stdenv.isDarwin))
```
Comment (RaitoBezarius) [inline pkgs/tools/system/netdata/default.nix:11]: This makes no sense, please simplify the boolean condition:
```suggestion
, withCloud ? false
```

you should put `stdenv.isDarwin` somewhere **else**

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -10229,6 +10229,10 @@ with pkgs;
...
```
Comment (RaitoBezarius) [inline pkgs/top-level/all-packages.nix:10233]: The conditional should go *here*:
```suggestion
    withCloud = !stdenv.isDarwin;
```

Comment (RaitoBezarius) [reply pkgs/top-level/all-packages.nix:10233]: Or you should guard netdataCloud behind an assert.

Comment (Izorkin) [reply pkgs/top-level/all-packages.nix:10233]: Thanks, fixed.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -10229,6 +10229,10 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation IOKit;
     protobuf = protobuf3_21;
   };
+  netdataCloud = netdata.override {
+    withCloud = !stdenv.isDarwin;;
```
Comment (RaitoBezarius) [inline pkgs/top-level/all-packages.nix:10233]: syntax error

Comment (Izorkin) [reply pkgs/top-level/all-packages.nix:10233]: Fixed :)

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**